otter/test/convergence/test_gathering.py

```diff
@@ -220,28 +220,23 @@ def setUp(self):
             self, 'otter.convergence.gathering.extract_CLB_drained_at',
             side_effect=lambda f: self.feeds[f])
 
-    def _request(self):
-        def request(service_type, method, url, json_response=False):
-            assert service_type is ServiceType.CLOUD_LOAD_BALANCERS
-            response = self.reqs[(method, url, json_response)]
-            return Effect(Stub(Constant(response)))
-        return request
-
-    def _resolve_retry_stubs(self, eff):
+    def _resolve_request(self, eff):
         """
-        Like :func:`resolve_retry_stub`, but assert what we expect to be the
-        correct policy.
+        Resolve a :obj:`ServiceRequest` based on ``self.reqs`` and assert
+        that it's wrapped in a Retry with the expected policy.
         """
         self.assertEqual(
             eff.intent.should_retry,
             ShouldDelayAndRetry(can_retry=retry_times(5),
                                 next_interval=exponential_backoff_interval(2)))
-        return resolve_retry_stubs(eff)
+        req = eff.intent.effect.intent
+        response = self.reqs[req.method, req.url, req.json_response]
```

Comment:
Hmm. Didn't know you can key on a tuple like this. Do you think it would be better to have it like `self.reqs[(req.method, req.url, req.json_response)]`? Non-blocking nitpick of course :)